UnitTests/BraintreeCoreTests/BTGraphQLHTTP_Tests.swift

```diff
@@ -0,0 +1,540 @@
+import XCTest
+import OHHTTPStubs
+import BraintreeTestShared
+@testable import BraintreeCore
+
+final class BTGraphQLHTTP_Tests: XCTestCase {
+
+    var http: BTGraphQLHTTP?
+
+    var fakeSession: URLSession {
+        let testConfiguration = URLSessionConfiguration.ephemeral
+        testConfiguration.protocolClasses = [BTHTTPTestProtocol.self]
+        return URLSession(configuration: testConfiguration)
+    }
+
+    override func setUp() {
+        http = BTGraphQLHTTP(url: BTHTTPTestProtocol.testBaseURL(), authorizationFingerprint: "test-authorization-fingerprint")
+    }
+
+    override func tearDown() {
+        HTTPStubs.removeAllStubs()
+    }
+
+    // MARK: - Basic request handling
+
+    func testRequests_useTheHostAtTheBaseURL() {
+        let expectation = expectation(description: "GET callback")
+        http?.session = fakeSession
+
+        http?.post("") { body, _, _ in
+            let httpRequest = BTHTTPTestProtocol.parseRequestFromTestResponseBody(body!)
+
+            XCTAssertTrue(httpRequest.url!.absoluteString.contains("bt-http-test://base.example.com:1234/base/path"))
+            expectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 2)
+    }
+
+    func testRequests_ignoreThePath() {
+        let expectation = expectation(description: "GET callback")
+        http?.session = fakeSession
+
+        http?.post("hey/go/here.html") { body, _, _ in
+            let httpRequest = BTHTTPTestProtocol.parseRequestFromTestResponseBody(body!)
+
+            XCTAssertEqual(httpRequest.url!.absoluteString, "bt-http-test://base.example.com:1234/base/path")
+            expectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 2)
+    }
+
+    // MARK: - Unsupported requests
+
+    func testGETRequests_areUnsupported() {
+        do {
+            try BTExceptionCatcher.catchException {
+                self.http?.get("") { _, _, _ in
```

Comment:
random q: can we remove these methods since all they do is throw?

Reply:
Do you mean the test methods or the actual method overrides in BTGraphQLHTTP?